File: 1013-FibonacciNumber/1013-FibonacciNumber.js
// Last updated: 24/07/2026, 23:50:02
/**
 * @param {number} n
 * @return {number}
 */
var fib = function (n, memo = {}) {
    if (n in memo) {
        return memo[n]
    }
    if (n <= 1) return n;
    memo[n] = fib(n - 1, memo) + fib(n - 2, memo);
    return memo[n]
};

File: 1013-FibonacciNumber/1013-FibonacciNumber.java
// Last updated: 24/07/2026, 23:50:08
class Solution {
    public int fib(int n) {
        if(n==0||n==1){
            return n; 
        }

        int fib1=fib(n-1);
        int fib2=fib(n-2);

        int ans = fib1+fib2;

        return ans;
    }
}